src/PrefixingContainer.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Dhii\Container;
+
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+/**
+ * A container implementation that wraps around an inner container and prefixes its keys, requiring consumers to
+ * include them when fetching or looking up data.
+ *
+ * @since [*next-version*]
+ */
+class PrefixingContainer implements ContainerInterface
+{
+    /**
+     * @since [*next-version*]
+     *
+     * @var ContainerInterface
+     */
+    protected $inner;
+
+    /**
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $prefix;
+
+    /**
+     * @since [*next-version*]
+     *
+     * @var bool
+     */
+    protected $strict;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param ContainerInterface $container The container whose keys to prefix.
+     * @param string             $prefix    The prefix to apply to the container's keys.
+     * @param bool               $strict    Whether or not to fallback to un-prefixed keys if a prefixed key does not
+     *                                      exist in the inner container.
+     */
+    public function __construct(ContainerInterface $container, string $prefix, bool $strict = true)
```

Comment:
IMHO it would be more intuitive and less prone to human error if the default value of `$strict` would be `false`.

Reply:
You'd allow access to un-prefixed keys by default?
I honestly don't have an opinion on this. I thought I did, but I started writing this reply :rofl: It sounds like a very subjective thing TBH

Reply:
Well, my opinion stems from the imaginary case where you say "I want a container with these keys prefixed". What would you intuitively imply about the other keys?

Reply:
> "I want a container with these keys prefixed" ... What would you intuitively imply about the other keys?

But this container does not allow you to choose which keys get prefixed. It's all or nothing.